Ignore Deployment Restriction for WebSocket 2.1,  Added websocket deployment tests

isaacrivriv opened the pull request:
- Made changes to include functionality to ignore restriction of adding endpoints after deployment
- Added test to verify deployment restrictions

fixes #22364

isaacrivriv committed:
Added websocket deployment test
- Made changes to include functionality to ignore restriction of adding endpoints after deployment
- Added test to verify deployment restrictions

isaacrivriv commented:
#build

volosied reviewed (dismissed):
Looks good to me. Thanks for getting this up!

pnicolucci commented on dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/BasicTest.java:
```diff
@@ -216,6 +224,30 @@ public void testURIOverlappingWithHtml() throws Exception {
+        // This test causes SRVE0777E: Exception thrown by application class ServerContainerExt.addEndpoint which is expected
+        // Also cause SRVE0315E An exception occurred: java.lang.Throwable: java.lang.IllegalStateException: endpoint.addsclosed also expected
+        // and SRVE8094W WARNING: Cannot set header. Response already committed to show up
+        if(RepeatTestFilter.isRepeatActionActive(JakartaEE10Action.ID)){
```
Any reason you didn't just do `JakartaEE10Action.isActive()` ?

isaacrivriv replied:
No specific reason, I was working with the RepeatTestFilter lately and just went with it here. I can change it back to make it consistent across all the tests

isaacrivriv replied:
Changed so that it matches the other tests we've worked on

pnicolucci commented on dev/io.openliberty.wsoc.internal_fat/test-applications/basic.war/src/basic/war/servlet/AddEndpointServlet.java:
```diff
@@ -0,0 +1,76 @@
+ * be relaxed in a future version."
+ * 
+ * Before 2.1, endpoints can only registered during the deployment of the web application. Once a websocket request enters the webcontainer,
+ * no more endpoints could be added with the API as stated in the spec "are only operational during the application deployment phase of an application. Specifically, as soon as any of the server endpoints within the application have accepted an opening handshake request, the apis may not longer be used. This restriction may be relaxed in a future version.". This restriction was removed in EE10 We ...
```
Can you do this line over multiple lines? It seems super long relative to the rest of the javadoc

isaacrivriv replied:
Sure no prob I'll make the change

isaacrivriv committed:
Fixed review comments
- Added spacing for javadoc
- Updated EE10 check to use the same as our other tests

isaacrivriv commented:
#build

pnicolucci reviewed (approved):
Looks good, thanks for making the requested changes!

isaacrivriv commented:
#libby
